qiskit/aqua/algorithms/amplitude_estimators/ae.py

```diff
@@ -122,14 +184,21 @@ def _evaluate_statevector_results(self, probabilities: Union[List[float], np.nda
             probabilities: The probabilities obtained from the statevector simulation,
                 i.e. real(statevector * statevector.conj())[0]
 
+        Raises:
+            AquaError: If `self._circuit` is not set.
+
         Returns:
             Dictionaries containing the a gridpoints with respective probabilities and
                 y measurements with respective probabilities, in this order.
         """
+        if self._circuit is None:
+            raise AquaError('Before calling _evaluate_statevector_results, _circuit must be set. '
```

Comment:
This error message again refers to calling private methods and private variables. Maybe it suffices to state that construct_circuit() must have be called to use.  I see this is a docstring on private method that the user should not be calling. If an end user ever saw this message it would be due to some internal programming error on the part of the algorithm right and not due to some incorrect sequence the user did? As such this is more an Internal Logic Error of the algorithm that presumably only the algo developer can fix?

Reply:
Yes, if users only call public methods this error only arises if something internal is broken. I think we have a few places where this "internal logic error" is the case. Would it make sense to note this somewhere? 
Now I'll just put that `constuct_circuit` must be called prior to this method because it sets the internal `_circuit` variable to give reasonable information to both the end user and the developer.

Reply:
Updated in 1def8c675